src/sage/topology/moment_angle_complex.py

```diff
@@ -814,3 +819,712 @@ def has_trivial_lowest_deg_massey_product(self):
+                -h^{6,0}
+                sage: b * a
+                h^{6,0}
+                sage: c.get_simplicial_complex()
+                Simplicial complex with vertex set (1, 2, 3, 4) and
+                facets {(1, 2), (1, 4), (2, 3), (3, 4)}
+
+                sage: Z = MomentAngleComplex([[1, 2], [1, 4], [2, 3], [3, 5],
+                ....:                         [5, 6], [3, 4], [2, 6], [4, 6]])
+                sage: H = Z.cohomology_ring(GF(2))
+                sage: a = H.basis()[4, 0]; a.get_simplicial_complex()
+                Simplicial complex with vertex set (1, 2, 5) and facets {(5,), (1, 2)}
+                sage: b = H.basis()[4, 1]; b.get_simplicial_complex()
+                Simplicial complex with vertex set (1, 3, 5) and facets {(1,), (3, 5)}
+
+            Here the vertex sets are not disjoint, so the
+            cup product is trivial. Because of this definition,
+            multiplying any cohomology class with itself yields 0::
+
+                sage: a * b
+                0
+                sage: all(a ** n == H.zero() for n in range(2, dim(Z)))
+                True
+            """
+            return self * other
+
+
+# Used for computing coeffeicients when multiplying in cohomology
+def eps(subcomplex, simplicial_complex):
+    r"""
+    Return the coefficient `\epsilon`, used when computing
+    the cup product of cohomology classes.
+
+    By definition `\epsilon(j, J) = (-1)^{r-1}`, where `j`
+    is the `r`-th element of `J`. For a subset `L\subset J`
+    we define
+
+    .. MATH::
+
+        \epsilon(L, J) := \prod_{j \in L} \epsilon(j, J).
+
+    INPUT:
+
+    - ``subcomplex`` -- a set or a simplicial complex,
+      represents `L` given above
+    - ``simplicial_complex`` -- a set or a simplicial
+      complex, represents `J` given above
+
+    REFERENCES:
+
+        Fore more information, see [Lin2019]_, p. 13.
+
+    EXAMPLES::
+
+        sage: from sage.topology.moment_angle_complex import eps
+        sage: eps({1,2}, {1,2,3,4})
+        -1
+        sage: eps({1,3}, {1,2,3,4})
+        1
+        sage: eps({2}, SimplicialComplex([[1,2], [2,3,4], [4,1]]))
+        -1
+    """
+    def _eps(element, simplicial_complex):
+        if element not in simplicial_complex._vertex_to_index:
+            raise ValueError("{} is not a vertex of this simplicial complex".format(element))
+        return (-1) ** simplicial_complex._vertex_to_index[element]
+
+    if not isinstance(subcomplex, SimplicialComplex):
+        # This may completely change the structure,
+        # but the ordering of the vertices will
+        # remain the same, and that's the only things
+        # that matters
+        subcomplex = SimplicialComplex([subcomplex])
+    if not isinstance(simplicial_complex, SimplicialComplex):
+        simplicial_complex = SimplicialComplex([simplicial_complex])
+    res = 1
+    for element in subcomplex.vertices():
+        res *= _eps(element, simplicial_complex)
+    return res
```

Comment:
I don't see the need for the `_eps` method. I would just put it inline here. This can also be simplified
```suggestion
    return (-1) ** sum(simplicial_complex._vertex_to_index[element]
                       for element in subcomplex.vertices())
```